### PR TITLE
fix: Update PresenceBadge to use resizeable icons

### LINKED ISSUE #20650: PresenceBadge: Missing presence icon sizes
Follow up from #20627

Since the presence badges are implemented fully as icons in the system icons package (@fluentui/react-icons), there are missing sizes from the design ramp which are:
* tiny - 6px
* ~large - 20px~
* ~extra-large - 28px~

If resizable icons become a reality, this should be easily mitigated

### LINKED ISSUE #29007: PresenceBadge: Update to use new icon sizes
This PR is to track adding the size 28 PresenceBadge icon

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR is to update the icons that `PresenceBadge` uses to the resizeable versions. 

Fixes #20650 
Fixes #29007

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Avatar-mount_base.html">652</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Avatar-mount.html">600</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Button-mount_base.html">326</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Button-mount.html">323</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Field-mount_base.html">1099</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Field-mount.html">1097</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProvider-mount_base.html">672</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProvider-mount.html">694</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-mount.html">85</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">60</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/MakeStyles-mount_base.html">869</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/MakeStyles-mount.html">869</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Persona-mount_base.html">1743</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/Persona-mount.html">1717</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/SpinButton-mount_base.html">1392</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29288/perf-test-react-components/SpinButton-mount.html">1370</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                            |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                           Change |
| :----------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                      | `84.006 kB`<br />`23.008 kB` | `78.196 kB`<br />`21.354 kB` |  `-5.81 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-1.654 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                   | `49.384 kB`<br />`15.546 kB` | `43.574 kB`<br />`13.882 kB` |  `-5.81 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-1.664 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr> | `64.036 kB`<br />`19.896 kB` | `58.226 kB`<br />`18.217 kB` |  `-5.81 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-1.679 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>      |  `24.393 kB`<br />`8.893 kB` |  `18.592 kB`<br />`7.254 kB` | `-5.801 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-1.639 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                | `56.275 kB`<br />`17.425 kB` | `50.465 kB`<br />`15.775 kB` |   `-5.81 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /><br />`-1.65 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/decrease.png" /> |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `18.782 kB`<br />`7.48 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `25.987 kB`<br />`8.384 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `26.888 kB`<br />`8.687 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.572 kB`<br />`19.658 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.688 kB`<br />`59.552 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `40.966 kB`<br />`13.569 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.541 kB`<br />`2.227 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `157.087 kB`<br />`43.832 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.017 kB`<br />`13.414 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `129.841 kB`<br />`34.929 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `75.114 kB`<br />`20.231 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `73.745 kB`<br />`19.826 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr>                                                              |   `14.356 kB`<br />`5.746 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                                                                    |   `29.146 kB`<br />`9.108 kB` |
| <samp>react-tags-preview</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>                                                                          |  `72.793 kB`<br />`21.685 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/dbf9aca4bfb9c2cb89d571f441152410a7b60af0'>dbf9aca4bfb9c2cb89d571f441152410a7b60af0</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-badge-e7a2177f-7d2a-4f6e-9312-074b3a71b06d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-badge-e7a2177f-7d2a-4f6e-9312-074b3a71b06d.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29288&commitId=ea8969d3b53fa5d1fe6b4b38dc09f9833bffbd3d&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 7 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.customSize+name+badge.normal.chromium.png|2| Changed|
|Avatar Converged.customSize+icon+active.normal.chromium.png|2| Changed|
|Avatar Converged.size+icon+badge+square.normal.chromium.png|28| Changed|
|Avatar Converged.size+inactive+badge.normal.chromium.png|14| Changed|
|Avatar Converged.badgeMask.normal.chromium.png|29| Changed|
|Avatar Converged.size+image+badge.normal.chromium.png|8| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|30| Changed|

</details>


  <details><summary><b>Persona Converged</b> 7 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Persona Converged.basic - High Contrast.normal.chromium.png|4| Changed|
|Persona Converged.basic - RTL.normal.chromium.png|2| Changed|
|Persona Converged.basic - Dark Mode.normal.chromium.png|1| Changed|
|Persona Converged.basic.normal.chromium.png|2| Changed|
|Persona Converged.textPosition.normal.chromium.png|6| Changed|
|Persona Converged.textAlignment.normal.chromium.png|4| Changed|
|Persona Converged.size-presence.normal.chromium.png|5| Changed|

</details>


  <details><summary><b>PresenceBadge Converged - OOF status</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - OOF status.default - RTL.chromium.png|83| Changed|
|PresenceBadge Converged - OOF status.default.chromium.png|83| Changed|

</details>


  <details><summary><b>PresenceBadge Converged - inverted background</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - inverted background.default.default.chromium.png|446| Changed|
|PresenceBadge Converged - inverted background.default - Dark Mode.default.chromium.png|381| Changed|
|PresenceBadge Converged - inverted background.default - High Contrast.default.chromium.png|356| Changed|

</details>


  <details><summary><b>PresenceBadge Converged - sizes</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - sizes.default - RTL.chromium.png|5| Changed|
|PresenceBadge Converged - sizes.default.chromium.png|5| Changed|

</details>


  <details><summary><b>PresenceBadge Converged - status</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|PresenceBadge Converged - status.default.chromium.png|43| Changed|
|PresenceBadge Converged - status.default - RTL.chromium.png|43| Changed|

</details>


  <details><summary><b>Table layout flex</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout flex.multiselect (checked) - neutral - High Contrast.chromium.png|15| Changed|
|Table layout flex.multiselect (mixed) - neutral - High Contrast.chromium.png|9| Changed|
|Table layout flex.single select (checked) - neutral - High Contrast.chromium.png|3| Changed|

</details>


  <details><summary><b>Table layout flex - cell actions</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout flex - cell actions.default - High Contrast.hover row.chromium.png|3| Changed|
|Table layout flex - cell actions.always visible - High Contrast.hover row.chromium.png|3| Changed|

</details>


  <details><summary><b>Table layout table</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout table.multiselect (checked) - neutral - High Contrast.chromium.png|15| Changed|
|Table layout table.multiselect (mixed) - neutral - High Contrast.chromium.png|9| Changed|
|Table layout table.single select (checked) - neutral - High Contrast.chromium.png|3| Changed|

</details>


  <details><summary><b>Table layout table - cell actions</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Table layout table - cell actions.always visible - High Contrast.hover row.chromium.png|3| Changed|
|Table layout table - cell actions.default - High Contrast.hover row.chromium.png|3| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.styles.ts [hunk 2]
```diff
@@ -71,32 +71,50 @@ const useStyles = makeStyles({
...
     '& svg': {
-      width: '6px !important',
-      height: '6px !important',
+      fontSize: '6px',
+    },
+  },
+  'extra-small': {
+    aspectRatio: '1',
+    width: '10px',
+    backgroundClip: 'unset', // tiny icons have a border less than 1px wide, and can't use the backgroundClip fix
```
Comment (behowell) [inline packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.styles.ts:85]: Was this intended to be copied to sizes other than tiny?
```suggestion
```

File: packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.styles.ts [hunk 3]
```diff
@@ -71,32 +71,50 @@ const useStyles = makeStyles({
...
+      fontSize: '10px',
+    },
+  },
+  small: {
+    aspectRatio: '1',
+    width: '12px',
+    backgroundClip: 'unset', // tiny icons have a border less than 1px wide, and can't use the backgroundClip fix
+    '& svg': {
+      fontSize: '12px',
+    },
```
Comment (behowell) [inline packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.styles.ts:96]: `fontSize` should work without the `'& svg'` selector. And in fact, I think you should be able to remove the `aspectRatio` and `width` as well, now that it's using resizable icons that respond to `fontSize`:
```suggestion
    fontSize: '12px',
```

Comment (behowell) [reply packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadgeStyles.styles.ts:96]: Ditto for all other sizes as well.

File: packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadge.tsx [hunk 4]
```diff
@@ -1,38 +1,39 @@
...
-  presenceAvailableRegular,
-  presenceAwayFilled,
-  presenceBlockedRegular,
-  presenceBusyFilled,
-  presenceDndFilled,
-  presenceDndRegular,
-  presenceOfflineRegular,
-  presenceOofRegular,
-  presenceUnknownRegular,
-} from './presenceIcons';
```
Comment (behowell) [inline packages/react-components/react-badge/src/components/PresenceBadge/usePresenceBadge.tsx:14]: Now that we're not using the definitions in `presenceIcons.ts`, we should add `@deprecated` comments to them. For example:
```ts
/** @deprecated Use PresenceAwayFilled from @fluentui/react-icons */
export const presenceAwayFilled = ...
```

File: packages/react-components/react-badge/src/index.ts [hunk 5]
```diff
@@ -5,16 +5,27 @@ export {
...
+  /** @deprecated */
   presenceDndRegular,
+  /** @deprecated */
   presenceDndFilled,
+  /** @deprecated */
   presenceOofRegular,
+  /** @deprecated */
   presenceOfflineRegular,
+  /** @deprecated */
   presenceUnknownRegular,
```
Comment (khmakoto) [inline packages/react-components/react-badge/src/index.ts:29]:
```suggestion
  // eslint-disable-next-line deprecation/deprecation
  presenceAwayRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceAwayFilled,
  // eslint-disable-next-line deprecation/deprecation
  presenceAvailableRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceAvailableFilled,
  // eslint-disable-next-line deprecation/deprecation
  presenceBlockedRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceBusyFilled,
  // eslint-disable-next-line deprecation/deprecation
  presenceDndRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceDndFilled,
  // eslint-disable-next-line deprecation/deprecation
  presenceOofRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceOfflineRegular,
  // eslint-disable-next-line deprecation/deprecation
  presenceUnknownRegular,
```

**Comment by khmakoto** [comment]
@tomi-msft it seems like you need to disable the eslint rule in the stories as well.

**Comment by tomi-msft** [comment]
After further investigation, the optimal path forward is not to replace the icons with the resizable versions, as the visual fidelity of the smaller sized `PresenceBadge` is too low when trying to resize it.

**Pull request closed without merging**